core/runtime/impl/runtime_external_interface.cpp

```diff
@@ -0,0 +1,288 @@
+        auto ptr = extension_->ext_get_allocated_storage(
+            arguments.at(0).geti32(), arguments.at(1).geti32(),
+            arguments.at(2).geti32());
+        return wasm::Literal(ptr);
+      }
+      /// ext_get_storage_into
+      if (import->base == ext_get_storage_into) {
+        auto res = extension_->ext_get_storage_into(
+            arguments.at(0).geti32(), arguments.at(1).geti32(),
+            arguments.at(2).geti32(), arguments.at(3).geti32(),
+            arguments.at(4).geti32());
+        return wasm::Literal(res);
+      }
+      /// ext_set_storage
+      if (import->base == ext_set_storage) {
+        extension_->ext_set_storage(
+            arguments.at(0).geti32(), arguments.at(1).geti32(),
+            arguments.at(2).geti32(), arguments.at(3).geti32());
+        return wasm::Literal();
+      }
+      /// ext_blake2_256_enumerated_trie_root
+      if (import->base == ext_blake2_256_enumerated_trie_root) {
+        extension_->ext_blake2_256_enumerated_trie_root(
+            arguments.at(0).geti32(), arguments.at(1).geti32(),
+            arguments.at(2).geti32(), arguments.at(3).geti32());
+        return wasm::Literal();
+      }
+      /// ext_storage_changes_root
+      if (import->base == ext_storage_changes_root) {
+        auto res = extension_->ext_storage_changes_root(
+            arguments.at(0).geti32(), arguments.at(1).geti32(),
+            arguments.at(2).geti32(), arguments.at(3).geti32());
+        return wasm::Literal(res);
+      }
+      /// ext_storage_root
+      if (import->base == ext_storage_root) {
+        extension_->ext_storage_root(arguments.at(0).geti32());
+        return wasm::Literal();
+      }
+
+      /// IO extensions
+
+      /// ext_print_hex
+      if (import->base == ext_print_hex) {
+        extension_->ext_print_hex(arguments.at(0).geti32(),
+                                  arguments.at(1).geti32());
+        return wasm::Literal();
+      }
+      /// ext_print_num
+      if (import->base == ext_print_num) {
+        extension_->ext_print_num(arguments.at(0).geti64());
+        return wasm::Literal();
+      }
+      /// ext_print_utf8
+      if (import->base == ext_print_utf8) {
+        extension_->ext_print_utf8(arguments.at(0).geti32(),
+                                   arguments.at(1).geti32());
+        return wasm::Literal();
+      }
+
+      /// Cryptographuc extensions
+
+      /// ext_blake2_256
+      if (import->base == ext_blake2_256) {
+        extension_->ext_blake2_256(arguments.at(0).geti32(),
+                                   arguments.at(1).geti32(),
+                                   arguments.at(2).geti32());
+        return wasm::Literal();
+      }
+      /// ext_ed25519_verify
+      if (import->base == ext_ed25519_verify) {
+        auto res = extension_->ext_ed25519_verify(
+            arguments.at(0).geti32(), arguments.at(1).geti32(),
+            arguments.at(2).geti32(), arguments.at(3).geti32());
```

Comment:
what if there will be no 0..3 argument? exception will be thrown

Reply:
I think we should terminate the program in that case. The number of arguments is defined in wasm code. If it is not as in C++, then our program should not work

Reply:
Added check argument check for that